packages/flutter_tools/lib/src/base/error_handling_io.dart

```diff
@@ -588,6 +652,21 @@ class ErrorHandlingProcessManager extends ProcessManager {
 
   final ProcessManager _delegate;
   final Platform _platform;
+  static const _ProcessDelegate _processDelegate = _ProcessDelegate();
+  static bool _skipCommandLookup = false;
+
+  /// Bypass package:process command lookup for all functions in this block.
+  ///
+  /// This required that the fully resolved executable path is provided.
+  static Future<T> skipCommandLookup<T>(Future<T> Function() operation) async {
```

Comment:
Not sure if this is feasible, but could you just add new methods to `ErrorHandlingProcessManager` like `runWithoutLookup`, `startWithoutLookup` etc.

Another option would be to add a `lookup` optional named argument to `run`, `start`, etc., that defaults to `true`, which you could ignore if `_delegate` is a fake/mock.

Reply:
We don't always injected the error handling process manager though. I could `is` check and conditionally call, but I didn't really like that either.

> Another option would be to add a lookup optional named argument to run, start, etc., that defaults to true, which you could ignore if _delegate is a fake/mock

Same problem, if I change the interface I need to is/cast or rewrite large swaths of tests. It isn't practical.

Reply:
Got it.

One more thought. Instead of checking `_delegate is LocalProcessManager` below in each call, could you check it before setting `_skipCommandLookup = true` here?

Reply:
Ahh, good idea :)

Reply:
oh wait, no its static

Reply:
Not sure if it's a cleanup anymore, but `skipCommandLookup` could maybe take the instance that's about to be used as an argument.

Reply:
this doesn't work because some classes don't have a ProcessManager, they have a ProcessUtils